File: DOCE_CODE_V3_LLM/test_examples/test_all_languages.cpp
/**
 * C++ example
 */

#include <string>

void simpleFunction() {
    // Simple function
}

class TestClass {
public:
    /**
     * Constructor
     */
    TestClass(const std::string& name) : name_(name) {}

    /**
     * Return a greeting
     */
    std::string greet() {
        return "Hello, " + name_ + "!";
    }

private:
    std::string name_;
};

File: DOCE_CODE_V3_LLM/test_examples/test_all_languages.py
"""Python example"""

def simple_function():
    """A simple Python function"""
    return "Hello from Python :D"

class TestClass:
    def __init__(self, name: str):
        """Constructor"""
        self.name = name

    def greet(self) -> str:
        """Return a greeting"""
        return f"Hello, {self.name}!"
